synapse/rest/client/account.py

```diff
@@ -135,31 +125,41 @@ async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
             # Have the configured identity server handle the request
             ret = await self.identity_handler.requestEmailToken(
                 self.hs.config.registration.account_threepid_delegate_email,
-                email,
-                client_secret,
-                send_attempt,
-                next_link,
+                body.email,
+                body.client_secret,
+                body.send_attempt,
+                body.next_link,
             )
         else:
             # Send password reset emails from Synapse
             sid = await self.identity_handler.send_threepid_validation(
-                email,
-                client_secret,
-                send_attempt,
+                body.email,
+                body.client_secret,
+                body.send_attempt,
                 self.mailer.send_password_reset_mail,
-                next_link,
+                body.next_link,
             )
 
             # Wrap the session id in a JSON object
             ret = {"sid": sid}
 
         threepid_send_requests.labels(type="email", reason="password_reset").observe(
-            send_attempt
+            body.send_attempt
         )
 
         return 200, ret
 
 
+class PasswordBody(BaseModel):
```

Comment:
Not huge, but if these are used by a single servlet I think I'd make them inner classes?

Reply:
Probably; I toyed with the idea but wasn't sure which option was clearer to read.

Reply:
The nice thing about attaching it that way is you can eventually make it magic (e.g. something like if the base servlet finds `BODY_MODEL` on the servlet than you parse / validate it automatically and pass the parsed form into `on_GET`)

Reply:
That could well make sense.

I wouldn't be shocked to see `AccountValidityRenewServlet.PostBody` or something like that as a type.

We may want a few flavours as the bodies for POST and PUT and URL parameters are all different parts.

Reply:
I've tried this in 61f8f251d.